src/Maestro/maestro-angular/src/maestro-client/models.ts

```diff
@@ -1007,6 +1008,238 @@ export class BuildRef {
+            prBuildTime: number,
+            bestCaseTime: number,
+            worstCaseTime: number,
+            onLongestBuildPath: boolean,
+            goalTime: number,
+        }
+    ) {
+        this._id = id;
+        this._repository = repository;
+        this._branch = branch;
+        this._officialBuildTime = officialBuildTime;
+        this._prBuildTime = prBuildTime;
+        this._bestCaseTime = bestCaseTime;
+        this._worstCaseTime = worstCaseTime;
+        this._onLongestBuildPath = onLongestBuildPath;
+        this._goalTime = goalTime;
+    }
+
+    private _id: string;
+
+    public get id(): string {
+        return this._id;
+    }
+
+    private _repository: string;
+
+    public get repository(): string {
+        return this._repository;
+    }
+
+    private _branch: string;
+
+    public get branch(): string {
+        return this._branch;
+    }
+
+    private _officialBuildTime: number;
+
+    public get officialBuildTime(): number {
+        return this._officialBuildTime;
+    }
+
+    private _prBuildTime: number;
+
+    public get prBuildTime(): number {
+        return this._prBuildTime;
+    }
+
+    private _bestCaseTime: number;
+
+    public get bestCaseTime(): number {
+        return this._bestCaseTime;
+    }
+
+    private _worstCaseTime: number;
+
+    public get worstCaseTime(): number {
+        return this._worstCaseTime;
+    }
+
+    private _onLongestBuildPath: boolean;
+
+    public get onLongestBuildPath(): boolean {
+        return this._onLongestBuildPath;
+    }
+
+    public _goalTime: number;
+
+    public get goalTime(): number {
+        return this._goalTime;
+    }
+    
+    public isValid(): boolean {
+        return (
+            this._id !== undefined &&
+            this._officialBuildTime !== undefined
+        );
+    }
+
+    public static fromRawObject(value: any): FlowRef {
+        let result = new FlowRef({
+            id: value["id"] == null ? undefined : value["id"] as any,
+            repository: value["repository"] == null ? undefined : value["repository"] as any,
+            branch: value["branch"] == null ? undefined : value["branch"] as any,
+            officialBuildTime: value["officialBuildTime"] == null ? undefined : value["officialBuildTime"] as any,
+            prBuildTime: value["prBuildTime"] == null ? undefined : value["prBuildTime"] as any,
+            bestCaseTime: value["bestCasePathTime"] == null ? undefined : value["bestCasePathTime"] as any,
+            worstCaseTime: value["worstCasePathTime"] == null ? undefined : value["worstCasePathTime"] as any,
+            onLongestBuildPath: value["onLongestBuildPath"] == null ? undefined : value["onLongestBuildPath"] as any,
+            goalTime: value["goalTime"] == null ? undefined : value["goalTime"] as any,
+        });
+        return result;
+    }
+
+    public static toRawObject(value: FlowRef): any {
+        let result: any = {};
+        result["id"] = value._id;
```

Comment:
Why not 
```
return ({
  id: value._id,
  repository: value._repository,
   ...
})
```
Seems easier to parse than all the quoted property assignments.

Reply:
All of the stuff in maestro-client is generated code. This is just manually adding the new models.